enhancements/network/external-lb-vips.md

```diff
@@ -0,0 +1,242 @@
+Then the components which interact with networking will have to check this
+new field and handle this scenario appropriately.
+
+### Workflow Description
+
+1. The customer will configure their external loadbalancer(s), which includes configuring the VIPs
+   for API and Ingress and also configure the loadbalancer tool that is being used (e.g. HAproxy, F5, etc)
+   Note that we don't support Static IPs yet, so the deployment will have to configure the LB backends
+   with all the IPs that are available in the subnet of the control-plane machines.
+2. The customer will write the `install-config.yaml` file and set the new `loadBalancerType` platform specific field to `External`, and
+   will set the fields for `apiVIP` and `ingressVIP`.
+3. In the installer, if the platform does support external loadbalancer, some validation will be run
+   to check that the VIPs can already be resolved (the validation already exists for vsphere, we need to
+    make it re-usable). The new field for `LoadBalancerType` will be set in `PlatformSpec`.
+4. In machine-config-operator, if LB type is external, the internal loadbalancer will not be
+   deployed.
+
+### API Extensions
+
+A new field will be added to the `PlatformStatus` of the infrastructure object to tell
+the kind of the loadbalancer.
+
+```golang
+// PlatformLoadBalancerType defines the type of load balancer used by the cluster.
+// +kubebuilder:validation:Enum:="Internal";"External"
+type PlatformLoadBalancerType string
+const (
+  // LoadBalancerTypeExternal is a loadbalancer that is managed outside of the cluster by the customer.
+  LoadBalancerTypeExternal LoadBalancerType = "UserManaged"
+
+  // LoadBalancerTypeInternal is a loadbalancer that is managed by the OpenShift cluster.
+  LoadBalancerTypeInternal LoadBalancerType = "OpenShiftManaged"
+)
+
+// VSpherePlatformSpec holds the desired state of the vSphere infrastructure provider.
+type VSpherePlatformSpec struct {
+  // loadBalancerType defines if an internal or external load balancer
+  // is to be used for the openshift api and ingress endpoints.
+  // When set to Internal the static pods in charge of API and Ingress traffic load-balancing
+  // defined in the machine config operator will be deployed.
+  // When set to External these static pods will not be deployed and it is expected that
+  // the load balancer is configured out of band by the deployer.
+  // When omitted, this means no opinion and the platform is left to choose a reasonable default
+  // The current default value is Internal.
+  // +optional
+  LoadBalancerType PlatformLoadBalancerType `json:"loadBalancerType,omitempty"`
+}
+```
+
+This field will be set in the `install-config.yaml`:
+```yaml
+platform:
+  vsphere:
+    loadBalancerType: UserManaged
+```
+
+### Implementation Details/Notes/Constraints
+
+* CoreDNS will be deployed if external loadbalancer is being used.
+  However, it is still not supported to change the IPs for the API and Ingress VIPs
```

Comment:
This sentence can be a little confusing. Here's my suggestion:
```
However, it is not yet supported to change the DNS records for the API and Ingress VIPs in case the IPs have changed. This will be supported in the future.
```
Bonus point if you link to the enhancement proposal Ben mentioned.